### PR TITLE
ENH: Fully cover `pad` and backports from NumPy 0.10dev

### PR DESCRIPTION
**Pull request by JDWarner**
This is a long-in-waiting improvement to coverage for `pad`, bringing it to 100%, and it also includes improvements to the way arguments are parsed allowing floating point scalars or arrays to be correctly handled.

Essentially this just backports numpy/numpy#5361 to our in-house `pad`, to expose the full up to date functionality with NumPy compatibility back to 1.6.x

All changes are backward-compatible. Also, `six` is no longer required!

### CODE REVIEW

**Comment by JDWarner** [comment]
@coverage I can't wait to hear what you have to say about this.

File: skimage/util/tests/test_arraypad.py [hunk 1]
```diff
@@ -42,7 +85,7 @@ def test_check_maximum_1(self):
              99, 99, 99, 99, 99, 99, 99, 99, 99, 99,
              99, 99, 99, 99, 99,
 
-             0,   1,  2,  3,  4,  5,  6,  7,  8,  9,
+             0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
```
Comment (JDWarner) [inline skimage/util/tests/test_arraypad.py]: I feel like these lines are exposing something funny about GitHub's diff algorithm.

File: skimage/util/tests/test_arraypad.py [hunk 2]
```diff
@@ -86,11 +153,11 @@ def test_check_minimum_1(self):
         a = np.arange(100)
         a = pad(a, (25, 20), 'minimum')
         b = np.array(
-            [0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
-             0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
-             0,  0,  0,  0,  0,
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```
Comment (JDWarner) [inline skimage/util/tests/test_arraypad.py]: Here too. There is *nothing* different about the second-to-last zero in terms of character code.

File: skimage/util/tests/test_arraypad.py [hunk 3]
```diff
@@ -445,58 +792,58 @@ def test_check_large_pad(self):
...
-              7,  4,  5,  6,  7,  4,  5,  6,  7],
-             [10, 11,  8,  9, 10, 11,  8,  9, 10, 11, 8,  9, 10, 11,  8,  9, 10,
-              11,  8,  9, 10, 11,  8,  9, 10, 11],
-             [2,  3,  0,  1,  2,  3,  0,  1,  2,  3,  0,  1,  2,  3,  0,  1,  2,
-              3,  0,  1,  2,  3,  0,  1,  2,  3],
-             [6,  7,  4,  5,  6,  7,  4,  5,  6,  7,  4,  5,  6,  7,  4,  5,  6,
-              7,  4,  5,  6,  7,  4,  5,  6,  7],
-             [10, 11,  8,  9, 10, 11,  8,  9, 10, 11, 8,  9, 10, 11,  8,  9, 10,
-              11,  8,  9, 10, 11,  8,  9, 10, 11]]
+            [[10, 11, 8, 9, 10, 11, 8, 9, 10, 11, 8, 9, 10, 11, 8, 9, 10,
```
Comment (JDWarner) [inline skimage/util/tests/test_arraypad.py]: And now it's the third to last entry?

**Comment by blink1073** [comment]
Numpy and Travis agree that its a good fit, good enough for me, thanks @JDWarner!

**Pull request merged**